### DISCUSSION TITLE
How to use slither-check-erc in foundry project with mapping imports

### DISCUSSION THREAD

**Discussion by varunsrin** [crytic/slither#1727] [Q&A] [answered]
**Repro Steps**

Environment: OSX M1 with slither 0.9.2 and forge 0.2.0 (0453ac8 2023-02-28T00:14:39.246801Z)

1. Try to run slither-check-erc src/IdRegistry.sol IdRegistry on the [Farcaster Contracts repo](https://github.com/farcasterxyz/contracts)

```
slither.exceptions.SlitherError: Invalid compilation:
Invalid solc compilation Error: Source "openzeppelin/contracts/utils/Context.sol" not found: File not found. Searched the following locations: "".
 --> src/IdRegistry.sol:4:1:
  |
4 | import {Context} from "openzeppelin/contracts/utils/Context.sol";
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Error: Source "openzeppelin/contracts/access/Ownable.sol" not found: File not found. Searched the following locations: "".
 --> src/IdRegistry.sol:5:1:
  |
5 | import {Ownable} from "openzeppelin/contracts/access/Ownable.sol";
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Error: Source "openzeppelin-contracts/contracts/metatx/ERC2771Context.sol" not found: File not found. Searched the following locations: "".
 --> src/IdRegistry.sol:6:1:
  |
6 | import {ERC2771Context} from "openzeppelin-contracts/contracts/metatx/ERC2771Context.sol";
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 ```
 
Issue does not occur with the `slither .` command, which handles the imports correctly. Issue is resolved by changing the mapped imports to relative imports that do not use the mappings.

**Comment by 0xalpharush** [comment, marked as answer]
slither-check-erc uses two positional args: the first being the project directory and the second the contract name (`slither-check-erc project contractName`). I believe you should be able to run `slither-check-erc  . IdRegistry` to compile the project and then analyze `IdRegistry`. I would also recommend checking out our ERC20 properties that you can fuzz with Echidna and/or Foundry https://github.com/crytic/properties.

**Comment by varunsrin** [comment]
thanks, this resolved the issue!
